### PR TITLE
Subaru: read LKAS buttons

### PR DESCRIPTION
**Pull request by jnewb1**
**Description**
Starting to break up my [long PR](https://github.com/commaai/openpilot/pull/28403/files) into smaller pieces. This will log button presses on Subaru.

**Verification**
~~Should not modify any current behavior, will just be useful later.~~
I also cleaned up some preglobal stuff, so these changes will need to be verified.

depends: https://github.com/commaai/opendbc/pull/861

### CODE REVIEW

File: selfdrive/car/subaru/carstate.py [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
 from opendbc.can.can_define import CANDefine
 from common.conversions import Conversions as CV
 from selfdrive.car.interfaces import CarStateBase
 from opendbc.can.parser import CANParser
-from selfdrive.car.subaru.values import DBC, CAR, GLOBAL_GEN2, PREGLOBAL_CARS, SubaruFlags
+from selfdrive.car.subaru.values import DBC, CAR, GLOBAL_GEN2, PREGLOBAL_CARS, Buttons, SubaruFlags
+
+PREV_BUTTON_SAMPLES = 8
+
+def cruise_buttons_conversion(cruise_buttons):
```
Comment (sshane) [inline selfdrive/car/subaru/carstate.py:12]: Think the other ports use a dict constant in values.py or similar

File: selfdrive/car/subaru/carstate.py [hunk 2]
```diff
@@ -13,6 +39,8 @@ def __init__(self, CP):
     can_define = CANDefine(DBC[CP.carFingerprint]["pt"])
     self.shifter_values = can_define.dv["Transmission"]["Gear"]
 
+    self.cruise_buttons = deque([Buttons.NONE] * PREV_BUTTON_SAMPLES, maxlen=PREV_BUTTON_SAMPLES)
```
Comment (sshane) [inline selfdrive/car/subaru/carstate.py:42]: do you need a deque?

Comment (jnewb1) [reply selfdrive/car/subaru/carstate.py:42]: was copying hyundai, but i think we only need 1 sample really https://github.com/commaai/openpilot/blob/2ebd7ab088ade61bbf661c140483fc477d444bc2/selfdrive/car/hyundai/carstate.py#L22

File: selfdrive/car/subaru/carstate.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/subaru/carstate.py:12]: if we merge "Main", "Resume", and "Set" into a single DBC field then we could do that (like preglobal has right now)

**Pull request closed without merging**